### DISCUSSION TITLE
Menu / 'tooltip' like element inside drawer

### DISCUSSION THREAD

**Discussion by frankvp11** [zauberzeug/nicegui#3480] [Q&A] [answered]
### Question

I am looking to place a menu of sorts next to a left drawer, however for some reason it either places it within the drawer or doesn't show it at all. 
Ideally, it would look something like Discord's functionality when you click on a person (see attached screenshot). 
![image](https://github.com/user-attachments/assets/6c7ccd65-8fbd-4c8b-a326-313c1672c2a1)

Unfortunately, whenever I use code like this:
```py
from nicegui import ui

def open_menu():
    with ui.menu() as menu:
        ui.label("Settings")
        ui.menu_item("Option 1")
        ui.menu_item("Option 2")
        ui.menu_item("Close", menu.close)
        
with ui.left_drawer().style("background-color: lightgray") as drawer:
    button = ui.button("Click", on_click=open_menu )

ui.run()
```
It can't place it - it outputs this in the HTML: 
![image](https://github.com/user-attachments/assets/981ad681-c53f-4959-8411-e458cd46ac8d)

Using the anchor prop, I can get it to show up 
![image](https://github.com/user-attachments/assets/19b79567-1bf9-4015-89d6-f23a79d16f5f)
however as you can see in that screenshot, it places it in the top corner, not where I want it. 

Does anyone have any idea on how I can get it to spot I need it to be?

**Comment by python-and-novella** [comment]
Try this:
Popup Proxy | Quasar Framework
https://quasar.dev/vue-components/popup-proxy
```python3
from nicegui import ui
   
button = ui.button("Click")
with button:
    with ui.element('q-popup-proxy').props(":offset='[-80,-45]'"):
        with ui.card():
            ui.label("Settings")

ui.run()
```
![image](https://github.com/user-attachments/assets/12831071-ffbd-429a-8706-3e77d52b1f33)

**Comment by falkoschindler** [comment, marked as answer]
I was just about to propose something similar:
```py
with ui.left_drawer().classes('bg-gray-100'):
    with ui.button("Click"):
        with ui.menu().props('anchor="top right" self="top left"') as menu:
            ui.item("Settings").classes("text-bold")
            ui.menu_item("Option 1")
            ui.menu_item("Option 2")
            ui.menu_item("Close", menu.close)
```

![Screenshot 2024-08-12 at 13 53 23](https://github.com/user-attachments/assets/df27395d-d290-488c-8fe7-666ef1190dfb)

I'm not sure about the pros and cons of @python-and-fiction's `q-popup-proxy` though.

**Reply by frankvp11** [reply]
Is there a way to mark both things as an answer? Both work for me, but ultimately I think I'm going to go with @falkoschindler's answer. 
Thank you both though.

**Reply by python-and-novella** [reply]
`popup-proxy` is a `word-right` way to  design the element which is popuped.
you can do the same with `menu` .
```python3
button = ui.button("Click")
with button:
    with ui.menu().props(":offset='[-80,-45]'"):
        with ui.card():
            ui.label("Settings")
```
